### PR TITLE
[rllib] Eager execution for centralized critic example, fix simple optimizer for multiagent

### PR DESCRIPTION
**Pull request by ericl**
cc @eugenevinitsky

### CODE REVIEW

File: rllib/examples/custom_tf_policy.py [hunk 1]
```diff
@@ -20,13 +21,22 @@ def policy_gradient_loss(policy, model, dist_class, train_batch):
...
     return -tf.reduce_mean(
-        action_dist.logp(train_batch["actions"]) * train_batch["rewards"])
+        action_dist.logp(train_batch["actions"]) * train_batch["advantages"])
+
+
+def calculate_advantages(policy,
+                         sample_batch,
+                         other_agent_batches=None,
+                         episode=None):
+    sample_batch["advantages"] = discount(sample_batch["rewards"], 0.99)
```
Comment (richardliaw) [inline rllib/examples/custom_tf_policy.py]: this should try to fetch gamma from policy config instead right

File: rllib/optimizers/sync_samples_optimizer.py [hunk 2]
```diff
@@ -29,17 +32,22 @@ def __init__(self,
                  workers,
                  num_sgd_iter=1,
                  train_batch_size=1,
-                 sgd_minibatch_size=0):
+                 sgd_minibatch_size=0,
+                 standardize_fields=[]):
```
Comment (richardliaw) [inline rllib/optimizers/sync_samples_optimizer.py]: None

File: rllib/optimizers/sync_samples_optimizer.py [hunk 3]
```diff
@@ -29,17 +32,22 @@ def __init__(self,
...
         PolicyOptimizer.__init__(self, workers)
 
         self.update_weights_timer = TimerStat()
+        self.standardize_fields = standardize_fields
```
Comment (richardliaw) [inline rllib/optimizers/sync_samples_optimizer.py]: or []

**Review by richardliaw — APPROVED** [review]

File: rllib/examples/custom_tf_policy.py (hunk 1 above)
Comment (ericl) [reply rllib/examples/custom_tf_policy.py]: Not for the example.

File: rllib/optimizers/sync_samples_optimizer.py (hunk 3 above)
Comment (ericl) [reply rllib/optimizers/sync_samples_optimizer.py]: Changed to froenzeset

**Pull request merged**